### DISCUSSION TITLE
Improve documentation on themes, please

### DISCUSSION THREAD

**Discussion by jeffwright13** [extrawurst/gitui#942] [Q&A] [answered]
The default color scheme for gitui in iTerm2 is illegible on my Mac (iTmer2). The light cyan with yellow lettering can be almost impossible to see:
[hard-to-see](https://imgur.com/U7TCMv7)

I am trying to change my color scheme, but the theme file doesn't seem to have an entry for the cyan command blocks with yellow (if it is listed here, please let me know). It would be good to have all possible elements and their names documented.
[short-list](https://imgur.com/8qT7w63)

Finally, the [THEMES.md file](https://github.com/extrawurast/gitui/blob/master/THEMES.md) (as of 2021-06-29) incorrectly shows that the location of the theme config file (on MacOS) is $HOME/Library/Application Support/gitui/theme.ron. In fact, it is located at ~/.config/gitui. This is frustrating to a new user and in my case, was almost enough to make me uninstall. But the 'app' itself is 733t so if there is a way for me to surmout these issues, I would like to stay!

Macbook Pro 2012 Intel Catalina 10.15.7, 15" Retina, iTerm2 w/ "Bright Lights" color scheme applied.

**Comment by jeffwright13** [comment]
**update**
I changed my iTerm color scheme to "Dark Default" and this made the whole program easy to see. So perhaps A modified request would be to add a blurb in the README (or wherever) that suggests changing the terminal color scheme if things are not visible?

**Comment by horta** [comment]
Same here. I'm using Kitty terminal with gruvbox-like theme. Can't see the bottom options.
<img width="1474" alt="Screenshot 2021-07-08 at 09 51 38" src="https://user-images.githubusercontent.com/514522/124892796-185ff480-dfd2-11eb-8522-814022a92abf.png">

**Comment by extrawurst** [comment]
> Same here. I'm using Kitty terminal with gruvbox-like theme. Can't see the bottom options.

they use `command_fg` as foreground color and `selection_bg` as background color, see the [documentation](https://github.com/extrawurst/gitui/blob/master/THEMES.md) for more details

**Comment by extrawurst** [comment, marked as answer]
@jeffwright13 Since you are using iterm2 you will find your custom colors (using whatever theme you use) here:
<img width="1040" alt="Screenshot 2021-12-09 at 21 42 48" src="https://user-images.githubusercontent.com/776816/145472647-29fea4d9-902c-4793-81cd-f14b0f7833aa.png">

If you use `yellow` in gitui `theme.ron` it will pick up whatever color you have there and it is not under gitui's control. 

>  incorrectly shows that the location of the theme config file

fixed, thanks for pointing that out!

> was almost enough to make me uninstall

its tough to cater for everyone, I am sure you will find great alternatives out there, find the ones listed in the readme.

> A modified request would be to add a blurb in the README

you mean I should add detailed descriptions on how to use any possible terminal application available ? maybe you can make a PR for that

**Comment by adrian5** [comment]
Is there a list of available highlight targets, or are those in the default `theme.ron` the whole set?

```ron
(
    selected_tab: Reset,
    command_fg: White,
    command_bg: DarkGray,
    selection_bg: Blue,
    cmdbar_extra_lines_bg: Blue,
    disabled_fg: DarkGray,
    diff_line_add: Green,
    diff_line_delete: Red,
    diff_file_added: LightGreen,
    diff_file_removed: LightRed,
    diff_file_moved: LightMagenta,
    diff_file_modified: Yellow,
    commit_hash: Magenta,
    commit_time: LightCyan,
    commit_author: Green,
    danger_fg: Red,
    push_gauge_bg: Blue,
    push_gauge_fg: Reset,
)
```

**Reply by extrawurst** [reply]
all customisable components can be seen in `style.rs` in the `Theme` type. see https://github.com/extrawurst/gitui/blob/master/src/ui/style.rs#L308
